cmd/serve/sftp/sftp.go

```diff
@@ -110,3 +140,33 @@ sftp backend, but it may not be with other SFTP clients.
 		})
 	},
 }
+
+type StdioChannel struct {
+	stdio *restic.StdioConn
```

Comment:
I think using `restic.StdioConn` is overkill here - I think you can just define

```go
type StdioChannel struct {
	stdin  *os.File
	stdout *os.File
}

func (c *StdioChannel) Read(data []byte) (int, error) {
	return c.stdin.Read(data)
}

func (c *StdioChannel) Write(data []byte) (int, error) {
	return c.stdout.Write(data)
}

// Close closes both streams.
func (c *StdioChannel) Close() error {
	err1 := c.stdin.Close()
	err2 := c.stdout.Close()
	if err1 != nil {
		return err1
	}
	return err2
}
```

Reply:
Done.